pkg/asset/installconfig/azure/dns.go

```diff
@@ -76,32 +78,50 @@ func (config DNSConfig) GetPrivateDNSZoneID(rgName string, zoneName string) stri
 func (config DNSConfig) GetDNSZone() (*Zone, error) {
 	//call azure api using the session to retrieve available base domain
 	zonesClient := newZonesClient(config.Session)
-	allZones, _ := zonesClient.GetAllPublicZones()
-	if len(allZones) == 0 {
+	publicZonesMap, err := zonesClient.GetAllPublicZones()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve base domains")
+	}
+
+	if len(publicZonesMap) == 0 {
 		return nil, errors.New("no public dns zone found in your subscription")
 	}
-	zoneNames := []string{}
-	for zoneName := range allZones {
-		zoneNames = append(zoneNames, zoneName)
+
+	publicZones := make([]string, 0, len(publicZonesMap))
+	for name := range publicZonesMap {
+		for _, id := range publicZonesMap[name] {
+			parts := strings.Split(id, "/")
+			rgName := parts[4]
```

Comment:
nit: magic number `4` is a little disconcerting to me. Not sure what the solution is, perhaps a const REGION_NAME_INDEX or something with a comment describing it.

Reply:
I've documented how the string is split and what each index contains. Hopefully that is sufficient ;-)